tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -246,9 +248,21 @@ IotHttpsResponseHandle_t allocate_IotResponseHandle() {
     safeMalloc(sizeof(struct _httpsResponse));
   if(pResponseHandle) {
     size_t headerLen;
-    size_t bodyLen;
+    __CPROVER_assume(headerLen < CBMC_MAX_OBJECT_SIZE);
```

Comment:
Is this really needed here?  Can't this wait for the is_valid predicate?

Reply:
Without this check, the allocation goes through but header is neither NULL nor valid after the allocation. CBMC wasn't happy about it at the `valid_header_pointers` part the within is_valid. Please see this report: https://dtjjmrnini9ie.cloudfront.net/IotHttpsClient_AddHeader-20200826-191042/out/html/index.html

Do you think we should remove it from the is_valid predicate since it's redundant?